app/jobs/v3/delete_service_instance_job.rb

```diff
@@ -0,0 +1,93 @@
+require 'jobs/reoccurring_job'
+
+module VCAP::CloudController
+  module V3
+    class DeleteServiceInstanceJob < VCAP::CloudController::Jobs::ReoccurringJob
+      attr_reader :warnings
+
+      def initialize(guid, operation, audit_info)
+        super()
+        @service_instance_guid = guid
+        @operation = operation
+        @client_arguments = { accepts_incomplete: false }
+        @user_audit_info = audit_info
+        @warnings = []
+      end
+
+      def perform
+        client = VCAP::Services::ServiceClientProvider.provide({ instance: service_instance })
+
+        execute_request(client)
+
+        finish
+      end
+
+      def job_name_in_configuration
+        "service_instance_#{operation_type}"
+      end
+
+      def max_attempts
+        1
+      end
+
+      def resource_type
+        'service_instances'
+      end
+
+      def resource_guid
+        service_instance_guid
+      end
+
+      def display_name
+        "service_instance.#{operation_type}"
+      end
+
+      private
+
+      attr_reader :service_instance_guid
+
+      def service_instance
+        ManagedServiceInstance.first(guid: @service_instance_guid)
+      end
+
+      def execute_request(client)
+        broker_response = client.send(
```

Comment:
```suggestion
        broker_response = client.public_send(
```
Haven't had a look at how we use `send` elsewhere, but `public_send` makes sure you don't accidentally call private methods.